networkit/cpp/embedding/BiasedRandomWalk.hpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * BiasedRandomWalk.hpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from node2vec
+ *  part of snap [https://github.com/snap-stanford/snap]
+ *  Copyright (c) 2007-2019, Jure Leskovec (under BSD license)
+ *
+ *  see [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#ifndef BIASEDRANDOMWALK_HPP
+#define BIASEDRANDOMWALK_HPP
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+namespace Embedding {
+
+class BiasedRandomWalk {
+public:
+    using Walk = std::vector<node>;     // one walk
+    using AllWalks = std::vector<Walk>; // n walks for all nodes: 2 dimensions in one big chunk
+
+    /// preprocesses transition probabilities for random walks. Has to be called once before doWalks
+    /// calls
+    void preprocessTransitionProbs(double paramP, double paramQ);
+
+    /// Simulates walks from every node and writes it into walks vector
+    AllWalks doWalks(count walkLen, count numberOfWalks);
+
+    BiasedRandomWalk(const Graph &graph);
+
+private:
+    const Graph &graph;
```

Comment:
Probably it is better to keep a pointer to `graph` (as already done in `Node2Vec`).